build.assets/tooling/cmd/build-apt-repos/apt_repo_tool.go

```diff
@@ -0,0 +1,178 @@
+package main
+
+import (
+	"io/fs"
+	"log"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/gravitational/trace"
+	"github.com/sirupsen/logrus"
+)
+
+type AptRepoTool struct {
+	config       *Config
+	aptly        *Aptly
+	s3Manager    *s3manager
+	supportedOSs *map[string][]string
+}
+
+// Instantiates a new apt repo tool instance and performs any required setup/config.
+func NewAptRepoTool(config *Config, supportedOSs *map[string][]string) (*AptRepoTool, error) {
+	art := &AptRepoTool{
+		config:       config,
+		s3Manager:    NewS3Manager(*config.bucketName),
+		supportedOSs: supportedOSs,
+	}
+
+	aptly, err := NewAptly()
+	if err != nil {
+		return nil, trace.Wrap(err, "failed to create a new aptly instance")
+	}
+	art.aptly = aptly
+
+	return art, nil
+}
+
+// Runs the tool, performing all of the business logic outside of argument parsing.
```

Comment:
Nit: Instead of "all of the business logic", say what it actually does.

Reply:
Updated